tg/config.py

```diff
@@ -1,36 +1,58 @@
-import configparser
-import mailcap
-import mimetypes
+"""
+Every parameter (except for DEFAULT_CONFIG) can be
+overwritten by external config file
+"""
 import os
+import platform
+import runpy
 
-DEFAULT_CONFIG = os.path.expanduser("~/.config/tg/tg.conf")
+_os_name = platform.system()
+_darwin = "Darwin"
+_linux = "Linux"
+
+
+DEFAULT_CONFIG = os.path.expanduser("~/.config/tg/conf.py")
 DEFAULT_FILES = os.path.expanduser("~/.cache/tg/")
-max_download_size = "10MB"
-record_cmd = "ffmpeg -f avfoundation -i ':0' -ar 22050 -b:a 32k '{file_path}'"
-long_msg_cmd = "vim -c 'startinsert' {file_path}"
-editor = os.environ.get("EDITOR", "vi")
-
-
-def get_cfg(config=DEFAULT_CONFIG):
-    cfg = configparser.ConfigParser()
-    cfg.read(config)
-    return cfg
-
-
-def save_cfg(cfg, config=DEFAULT_CONFIG):
-    config_dir = os.path.dirname(config)
-    if not os.path.isdir(config_dir):
-        os.makedirs(config_dir)
-    with open(config, "w") as f:
-        cfg.write(f)
-
-
-def get_file_handler(file_name, default=None):
-    mtype, _ = mimetypes.guess_type(file_name)
-    if not mtype:
-        return default
-    caps = mailcap.getcaps()
-    handler, view = mailcap.findmatch(caps, mtype, filename=file_name)
-    if not handler:
-        return None
-    return handler
+LOG_LEVEL = "INFO"
+
+API_ID = "559815"
+API_HASH = "fd121358f59d764c57c55871aa0807ca"
+
+PHONE = None
+ENC_KEY = None
+
+TDLIB_PATH = None
+TDLIB_VERBOSITY = 0
+
+MAX_DOWNLOAD_SIZE = "10MB"
+
+# TODO: check platform
+NOTIFY_CMD = "/usr/local/bin/terminal-notifier -title '{title}' -subtitle '{subtitle}' -message '{msg}' -appIcon '{icon_path}'"
+# TODO: check platform
+if _os_name == _linux:
+    RECORD_CMD = "ffmpeg -f alsa -i default -ar 22050 -b:a 32k '{file_path}'"
+else:
+    RECORD_CMD = (
+        "ffmpeg -f avfoundation -i default -ar 22050 -b:a 32k '{file_path}'"
+    )
+
+# TODO: use mailcap instead of editor
+LONG_MSG_CMD = "vim -c 'startinsert' {file_path}"
+EDITOR = os.environ.get("EDITOR", "vi")
+
+if _os_name == _linux:
+    DEFAULT_OPEN = "xdg-open '{file_path}'"
+else:
+    DEFAULT_OPEN = "open '{file_path}'"
+
+if _os_name == _linux:
```

Comment:
idea: instead of checking for platform we could do something like this:
```
for tool in ['xclip, 'pbcopy, ...]:
  if exists(tool):
     DEFAULT_COPY = tool
     break
```
This could be useful for commands that could have possible a lot of tools.

Reply:
I don't want to support all existing tools in the world. We should support the default one for each platform, it used something different, user can define it yourself in config file.